app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -129,10 +131,26 @@ const Thread = ({author, channel, location, post, teammateNameDisplay, testID, t
     const textStyles = getMarkdownTextStyles(theme);
     const serverUrl = useServerUrl();
 
+    const [isChannelNamePressed, setIsChannelNamePressed] = useState<Boolean>(false);
+
+    const channelNameStyle = useMemo(() => (
+        [styles.channelName, isChannelNamePressed ? {color: theme.buttonBg} : null]
+    ), [isChannelNamePressed]);
+
+    const togglePressed = useCallback(() => {
+        setIsChannelNamePressed(!isChannelNamePressed);
```

Comment:
As my previous comment, if you use the function form, you can remove the dependency.